hphp/runtime/vm/jit/vasm-lower.cpp

```diff
@@ -33,6 +33,57 @@ namespace HPHP { namespace jit {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+namespace detail {
+
+jit::vector<VregConstraint> computeDominatingConstraints(Vunit& unit) {
+  jit::vector<VregConstraint> vregConstraints(unit.blocks.size(),
+                                              VregConstraint::Unknown);
+
+  // Build reverse postorder list of Vlabels
+  jit::vector<Vlabel> rpo;
+  PostorderWalker{unit}.dfs([&] (Vlabel ib) {
+    rpo.insert(rpo.begin(), ib);
+  });
+
+  // Calculate each block's dominating vreg constraint. The lowerers will use
+  // this as the initial vreg constraint for each block, adjusting as they
+  // encounter vregallow/vregrestrict during lowering.
+  vregConstraints[rpo[0]] = VregConstraint::Allow;
+  for (auto b: rpo) {
+    // Iterate through the instructions updating the constraint accordingly.
+    auto con = vregConstraints[b];
+    for (size_t i = 0; i < unit.blocks[b].code.size(); ++i) {
+      auto& inst = unit.blocks[b].code[i];
+      if (inst.op == Vinstr::vregrestrict) {
+        con = VregConstraint::Restrict;
+      } else if (inst.op == Vinstr::vregallow) {
+        con = VregConstraint::Allow;
+      }
+    }
+
+    // Pass the contraint to each of the successors, enforcing that a block
+    // isn't passed conflicting contraints from predecessors. i.e. A successor
+    // must be pass all restrict or all allow/none.
+    for (auto s: succs(unit.blocks[b])) {
```

Comment:
`for (auto s : succs(unit.blocks[b])) {`

(More generally, spaces around operators... and colons.)